### PR TITLE
remove astropy helpers, ah_bootstrap, setup.py; add pyprojet.toml

### PR DESCRIPTION
**Pull request by keflavich**
@betacygni @e-koch @astrofrog this could use some review b/c I haven't learned the packaging ecosystem, but currently statcont won't install on py3.12, so we need to do something here

### CODE REVIEW

File: setup.py
Comment (e-koch) [inline setup.py:1]: The `setup` call is missing here (e.g., https://github.com/radio-astro-tools/spectral-cube/blob/master/setup.py#L51)

Comment (keflavich) [reply setup.py:1]: I'm confused - I totally removed setup.py

Comment (keflavich) [reply setup.py:1]: you mean I should add it back with the setup call?

Comment (e-koch) [reply setup.py:1]: You still need it though? (Unless I'm also behind on the most recent packaging updates)

Comment (keflavich) [reply setup.py:1]: no, pip install works with just pyproject.toml.  But I don't know the reasons for/against keeping setup.py

**Comment by keflavich** [comment]
@smaret if you can confirm this solves the problem, we'll merge it.

File: setup.py
Comment (astrofrog) [reply setup.py:1]: Yes in fact you could even get rid of setup.cfg and just have pyproject.toml!

**Comment by smaret** [comment]
@keflavich Yes, my PR solved the installation, and I was also able to run the examples.

**Pull request merged**